core/src/main/java/fj/data/Stream.java

```diff
@@ -1554,9 +1548,7 @@ public A head() {
    * @param o The stream of streams to join.
    * @return A new stream that is the join of the given streams.
    */
-  public static <A> Stream<A> join(final Stream<Stream<A>> o) {
-    return Monoid.<A>streamMonoid().sumRight(o);
-  }
+  public static <A> Stream<A> join(final Stream<Stream<A>> o) { return o.bind(identity()); }
 
```

Comment:
There are two reasons:
* Performance: `sumRight()` calls strict `foldRight()` and hence requires with O(n) stack space. What is more important is that join will not work for infinite streams. On the other hand `bind() `calls lazy `foldRight`, which does not require O(n) stack space and is OK for infinite streams.
* Code conciness o.bind(identity) is shorter :)